### PR TITLE
fix(downloader): prevent open sockets

### LINKED ISSUE #665: investigate why recent epsagon client hangs
**Description**
Using [epsagon v1.58.5](https://github.com/epsagon/epsagon-node/releases/tag/v1.58.5) makes the pipeline hang **if** it requests `word2md` **and**  it is requested via `dispatch`.

eg invoking the `html` action directly works:

```
curl --request GET \
  --url 'https://adobeioruntime.net/api/v1/web/helix-pages/github-com--adobe--helix-pages--v1-8-14-dirty/html?owner=adobe&repo=theblog&path=%2Fen%2Farchive%2F2020%2Fintroducing-public-beta.md&ref=master'
```

but invoking via `dispatch` fails:

```
curl --request GET \
  --url 'https://adobeioruntime.net/api/v1/web/helix-pages/helix-services/dispatch@v3?content.owner=adobe&content.repo=theblog&content.ref=master&content.root=&content.index=index.html&content.package=github-com--adobe--helix-pages--v1-8-14-dirty&static.owner=adobe&static.repo=helix-pages&static.ref=master&static.root=%2Fhtdocs&path=%2Fen%2Farchive%2F2020%2Fintroducing-public-beta.html&rootPath=&params=&strain=default'
```

### PR DESCRIPTION
**Pull request by tripodsan**
fixes #665

### CODE REVIEW

**Comment by tripodsan** [comment]
@kptdobe this PR also fixes an error in the downloader such as the `x-request-id` was not set correctly. I think this is why the smoke-tests now fail for helix-cli. I will adjust the settings in helix-cli to exclude the `x-request-id` header for recording-matching (tomorrow)

File: src/utils/Downloader.js [hunk 1]
```diff
@@ -135,10 +134,11 @@ class Downloader {
       let res = {};
       try {
         res = await this._client(uri, options);
+        const body = await res.text();
```
Comment (rofe) [inline src/utils/Downloader.js:137]: Why introduce a variable?

Comment (stefan-guggisberg) [reply src/utils/Downloader.js:137]: to make sure the response body is always consumed, i.e. independent of `res.ok`.

Comment (rofe) [reply src/utils/Downloader.js:137]: Got it, thanks.

**Pull request merged**